bedrock/urls/mozorg_mode.py

```diff
@@ -52,3 +56,26 @@
         path("404/", import_string(handler404)),
         path("500/", import_string(handler500)),
     )
+
+if settings.WAGTAIL_ENABLE_ADMIN:
+    urlpatterns += (path("cms-admin/", include(wagtailadmin_urls)),)
+
+if settings.DEFAULT_FILE_STORAGE == "django.core.files.storage.FileSystemStorage":
+    # Serve media files from Django itself - production won't use this
+    from django.urls import re_path
+    from django.views.static import serve
+
+    urlpatterns += (
+        re_path(
+            r"^user-media/(?P<path>.*)$",
+            serve,
+            {"document_root": settings.MEDIA_ROOT},
+        ),
+    )
+    # Note that statics are handled via Whitenoise's middleware
+
+# Wagtail is the catch-all route, and it will raise a 404 if needed.
+# Note that we're also using localised URLs here
+urlpatterns += bedrock_i18n_patterns(
+    path("", include(wagtail_urls)),
+)
```

Comment:
This is important: Wagtail has one main view: `wagtail.views.serve` which must (via this `wagtail_urls` urls module) come at the end of all URLs, and does internal pattern matching based on the slug of a page (eg 'vpn-tips') combined with the slugs of any ancestors (eg 'vpn' <- 'products')

As such, wagtail_url will also raise a 404 if no match is found, so any custom 404 handling needs to bear this in mind. I had to minorly refactor how we do the special careers 404 page (which mentions if a job listing has disappeared) in light of this.